src/lib.rs

```diff
@@ -50,45 +50,43 @@ pub fn derive_deref_mut(input: TokenStream) -> TokenStream {
 fn parse_fields(item: &syn::DeriveInput, mutable: bool) -> (syn::Type, proc_macro2::TokenStream) {
     let trait_name = if mutable { "DerefMut" } else { "Deref" };
     let fields = match item.data {
-        syn::Data::Struct(ref body) => body.fields.iter().collect::<Vec<&syn::Field>>(),
+        syn::Data::Struct(ref body) =>
+            body.fields.iter()
+                .filter(|field| {
+                    if let Type::Path(TypePath { path: Path { segments, .. }, .. }) = &field.ty {
+                        let ident = &segments.last().expect("Expected path to have at least one segment").ident;
+                        ident != "PhantomData"
+                    } else {
+                        true
+                    }
+                })
+                .collect::<Vec<&syn::Field>>(),
         _ => panic!("#[derive({})] can only be used on structs", trait_name),
     };
 
-    let field_ty = match fields.len() {
-        1 => Some(fields[0].ty.clone()),
-        2 => {
-            if let Type::Path(TypePath { path: Path { segments, .. }, .. }) = &fields[1].ty {
-                if segments
-                    .last()
-                    .expect("Expected path to have at least one segment")
-                    .ident == "PhantomData"
-                {
-                    Some(fields[0].ty.clone())
-                } else {
-                    None
+    let target_field = match fields.len() {
+        1 => fields[0],
+        _ => {
+            // Look for attribute
+            let mut targets = Vec::new();
+            for field in fields.iter() {
+                if let Some(_) = field.attrs.iter().find(|attr| attr.path.is_ident("deref_target")) {
+                    targets.push(field);
                 }
-            } else {
-                None
             }
-        },
-        _ => None,
+            *targets.get(0).expect("#[deref_target] expected on one of the fields")
```

Comment:
This looks like a slient fail to me, as there should not be two such flags on a single structure.
This whole part (with the vec allocation & stuff) could be written:
```rust
let mut targets = fields.iter().filter(|field| field.attrs.iter().any(|attr| attr.path.is_ident("deref_target")));
match (targets.next(), targets.next()) {
    (Some(target), None) => target,
    (None, _) => panic!("#[derive({})]: since there is more than one field, #[deref_target] is needed on a field", trait_name),
    (Some(_), Some(_)) => panic!("#[derive({})]: only one #[deref_target] allowed", trait_name),
}
```
in order to not iterate on everything when possible, and to not allocate a targets `Vec` that seems unnecessary.

Reply:
https://github.com/Ploppz/derive_deref/pull/1